src/core/utils/geometryutils.cpp

```diff
@@ -38,6 +39,66 @@ QgsGeometry GeometryUtils::polygonFromRubberband( RubberbandModel *rubberBandMod
   return g;
 }
 
+QgsGeometry::OperationResult GeometryUtils::reshapeFromRubberband( QgsVectorLayer *layer, QgsFeatureId fid, RubberbandModel *rubberBandModel )
+{
+  QgsFeature feature = layer->getFeature( fid );
+  if ( feature.geometry().isNull() ||
+       ( QgsWkbTypes::geometryType( layer->wkbType() ) != QgsWkbTypes::LineGeometry && QgsWkbTypes::geometryType( layer->wkbType() ) != QgsWkbTypes::PolygonGeometry ) )
```

Comment:
Should we check layer of feature type here? I know this should currently match, but we never know for the future